Add information about adding CA cert when configuring LDAP (#5191)
theforeman/foreman-documentation: guides/common/modules/proc_obtaining-a-ca-certificate-for-ldaps.adoc

:_mod-docs-content-type: PROCEDURE

[id="obtaining-a-ca-certificate-for-ldaps"]
= Obtaining a CA certificate for LDAPS

[role="_abstract"]
If {Project} uses TLS to establish a secure LDAP connection (LDAPS), the CA certificates of your LDAP server must be trusted by {Project}.

.Prerequisites
* If your LDAP server uses a certificate chain with intermediate certificate authorities, the trusted CA list must contain all root and intermediate certificates.

.Procedure
* Obtain the CA certificate from the LDAP Server.

> 

Reviewer: ```suggestion
+
```

> If you use Active Directory Certificate Services, export the Enterprise PKI CA Certificate using the Base64 encoded X.509 format.

Reviewer 2: ```suggestion
If you use Active Directory Certificate Services, export the Enterprise PKI CA Certificate, using the Base64 encoded X.509 format.
```

ifndef::orcharhino[]

> For more information, see  https://access.redhat.com/solutions/1498773[How to configure Active Directory authentication with `TLS` on {Project}].

Reviewer 2: We want to hard-code the article title.
```suggestion
For more information, see  https://access.redhat.com/solutions/1498773[How to configure Active Directory authentication with TLS on Red Hat Satellite 6].
```

endif::[]

> 

Reviewer 2: ```suggestion
+
```

> 

Reviewer: Is that a prerequisite? If not, then maybe "Ensure that you provide the certificate in PEM format. {Project} does not accept binary formats." or similar.

:_mod-docs-content-type: PROCEDURE

[id="obtaining-a-ca-certificate-for-ldaps"]
= Obtaining a CA certificate for LDAPS

[role="_abstract"]
If {Project} uses TLS to establish a secure LDAP connection (LDAPS), the CA certificates of your LDAP server must be trusted by {Project}.

.Prerequisites
* If your LDAP server uses a certificate chain with intermediate certificate authorities, the trusted CA list must contain all root and intermediate certificates.

.Procedure
* Obtain the CA certificate from the LDAP Server.

If you use Active Directory Certificate Services, export the Enterprise PKI CA Certificate using the Base64 encoded X.509 format.
ifndef::orcharhino[]
For more information, see  https://access.redhat.com/solutions/1498773[How to configure Active Directory authentication with `TLS` on {Project}].
endif::[]

ifdef::foreman-deb[]
{Project} only accepts the `.crt` file extension for certificates in PEM ASCII format.
endif::[]
ifndef::foreman-deb[]
The filename extensions `.cer` and `.crt` are only conventions and can refer to DER binary or PEM ASCII format certificates.

> endif::[]

Reviewer 3: I don't think we need ifdefs here. If the user wants to supply a CA cert, it needs to be in PEM ASCII format

> endif::[]

Reviewer 3: I don't think we need ifdefs here. If the user wants to supply a CA cert, it needs to be in PEM ASCII format

Author: But, the first line says that `{Project} only accepts the `.crt` file extension`, however the second line mentions `.cer` and `.crt`. It'll sound contradictory if I remove the conditionals. 
Also, do we need to remove the reference to DER binary?

Reviewer 3: The file ~suffix~ extension is sort of irrelevant. You somehow obtain the cert, take its contents and set the contents as an attribute of the ldap auth source. We just need the thing you copy and paste to be in a specific format.

Reviewer 3: > Also, do we need to remove the reference to DER binary?

Yes, I would be fine even with just something like this.
```suggestion
{Project} only accepts certificates in PEM ASCII format.
```

Author: Thanks for the clarification!
